### DISCUSSION TITLE
Feature request: be able to digitally sign

### DISCUSSION THREAD

**Discussion by renis1235** [ZUGFeRD/mustangproject#519] [Ideas]
Hi,

I am writing my idea/ suggestion in English and not in German so that everyone can understand and give their opinion.
With that being said, I noticed that the Exporters (e.g. Implementations of `IZUGFeRDExporter`) do not support encrypted files.

As a Freelancer, I protect my Invoices from unauthorized changes and I think that improving the Exporters with the functionality to load protected files would make sense for future use.

I can contribute by opening a Pull Request when I am done with it.
I have some questions though:

1) Can I format the files as per Java-Convention? I noticed a couple of spaces and indents are not as usual formatted.
2) Are there any guidelines as to what must I do before opening a PR? Test-Coverage and so on...

I look forward to your input and answers.

Best Regards
RF

**Comment by winnyschuster** [comment]
Auszug aus der Factur-x/Zugferd Spezifikation (6.2)

- Das Dokument darf nicht verschlüsselt sein oder irgendeine Authentifizierung erforderlich machen. Die Verwendung von DRM (Digital Rights Management) ist unzulässig.

**Comment by heisej** [comment]
In my opinion PDF encryption will not be possible as the ZUGFeRD standard requires PDF/A files. And for those encryption is not allowed after all.

**Reply by J-N-K** [reply]
That does not rule out digitally signed PDFs, doesn't it? At least in PDF/A-3 this seems to be allowed.

**Reply by winnyschuster** [reply]
you were talking about encrypted files. Any kind of files which require some authorization are not allowed. Everybody that has access to the file must be allowed to see its contents

**Reply by renis1235** [reply]
@winnyschuster When restricting changes through a master password, everyone can see the document's contents, they just won't be able to make changes to the Document without the password.

Still, that does not bring anything, since that would make the Document non-PDF/A-compliant. ☹️

**Comment by renis1235** [comment]
Thank you for your replies and input.
I did not know that the ZUGFeRD Standard required PDF/A-compliant documents.
Then restricting changes through the means of encryption (Master Password) is not permitted.

Still, after my research, digitally signing a PDF would not encrypt it and would provide a means of ensuring the integrity of the document while being PDF/A-compliant.

But that would require signing the document *after* appending the XML.
